tests/Doctrine/Tests/ORM/Functional/Ticket/GH6141Test.php

```diff
@@ -0,0 +1,153 @@
+<?php
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Types\StringType;
+use Doctrine\DBAL\Types\Type;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+class GH6141Test extends OrmFunctionalTestCase
+{
+    protected function setUp()
+    {
+        parent::setUp();
+
+        Type::addType(GH6141PeopleType::NAME, __NAMESPACE__ . '\\GH6141PeopleType');
+
+        $this->_schemaTool->createSchema(array(
+            $this->_em->getClassMetadata(__NAMESPACE__ . '\\GH6141Person'),
+            $this->_em->getClassMetadata(__NAMESPACE__ . '\\GH6141Boss'),
+            $this->_em->getClassMetadata(__NAMESPACE__ . '\\GH6141Employee'),
+        ));
+    }
+
+    public function testEnumDiscriminatorsShouldBeConvertedToString()
+    {
+        $boss = new GH6141Boss('John');
+        $employee = new GH6141Employee('Bob');
+
+        $this->_em->persist($boss);
+        $this->_em->persist($employee);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $query = $this->_em->createQueryBuilder()
```

Comment:
`findOneBy()` instead - simpler here, as we don't go through the entire DQL scenarios